src/data/markdown/versioned-js-api/v0.38/javascript api/01 Init context/open.md

```diff
@@ -0,0 +1,123 @@
+---
+head_title: 'JavaScript API: open'
+title: 'open( filePath, [mode] )'
+description: 'Opens a file and reads all the contents into memory.'
+excerpt: 'Opens a file and reads all the contents into memory.'
+---
+
+Opens a file, reading all its contents into memory for use in the script.
+
+> #### Use [SharedArray](/javascript-api/k6-data/sharedarray/) for CSV and JSON files
+> `open()` often consumes a large amount of memory because every VU keeps a separate copy of the file in memory.
+> To reduce the memory consumption, we strongly advise the usage of [SharedArray](/javascript-api/k6-data/sharedarray/) for CSV, JSON and other files intended for script parametrization.
+
+<blockquote mod='warning'>
+
+#### Function only available in "init context"
+
+This is a function that can only be called from the init context (aka **init code**), code in the global context that is, outside of the main export default function { ... }.
+
+By restricting it to the init context, we can easily determine what local files are needed to run the test and thus what we need to bundle up when distributing the test to multiple nodes in a clustered/distributed test.
+
+See the example further down on this page. For a more in-depth description, see [Running k6](/getting-started/running-k6).
+
+#### Breaking change in v0.36.0
+
+Since k6 v0.36.0, VUs are now restricted to only `open()` files that were also opened in the [init context](https://k6.io/docs/using-k6/test-life-cycle/#init-and-vu-stages) of the first VU - the one that was initialized to get the exported `options` from the JS script (`__VU==0`). This means that the code like `if (__VU > 0) { const arr = open("./arr.json"); }` will result in an error.
```

Comment:
```suggestion
Since k6 v0.36.0, VUs are now restricted to only `open()` files that were also opened in the [init context](https://k6.io/docs/using-k6/test-life-cycle/#init-and-vu-stages) of the first VU, which was initialized to get the exported `options` from the JS script (`__VU==0`).
This means that the code like `if (__VU > 0) { const arr = open("./arr.json"); }` will result in an error.
```